core/Plugin.php

```diff
@@ -461,6 +463,38 @@ public function getMissingDependenciesAsString($piwikVersion = null)
         ));
     }
 
+    /**
+     * Schedules re-archiving of this plugin's reports from when this plugin was last
+     * deactivated to now. If the last time core:archive was run is earlier than the
+     * plugin's last deactivation time, then we use that time instead.
+     *
+     * Note: this only works for CLI archiving setups.
+     *
+     * Note: the time frame is limited by the `[General] rearchive_reports_in_past_last_n_months`
+     * INI config value.
+     *
+     * @throws \DI\DependencyException
+     * @throws \DI\NotFoundException
+     */
+    public function schedulePluginReArchiving()
+    {
+        $lastDeactivationTime = $this->getPluginLastDeactivationTime();
+
+        $dateTime = null;
+
+        $lastCronArchiveTime = (int) Option::get(CronArchive::OPTION_ARCHIVING_FINISHED_TS);
+        if (empty($lastCronArchiveTime)) {
+            $dateTime = $lastDeactivationTime;
+        } else if (empty($lastDeactivationTime)) {
+            $dateTime = $lastCronArchiveTime;
+        } else {
+            $lastCronArchiveTime = Date::factory($lastCronArchiveTime);
+            $dateTime = $lastDeactivationTime->isEarlier($lastCronArchiveTime) ? $lastDeactivationTime : $lastCronArchiveTime;
+        }
+
+        $archiveInvalidator = StaticContainer::get(ArchiveInvalidator::class);
```

Comment:
maybe we also add an `if (empty($dateTime))  { dateTime = null} ` just in case a value is false or 0?

Reply:
be also good to add this check not that we're passing like `$archiveInvalidator->scheduleReArchiving('all', $this->getPluginName(), $report = null, 0);` or anything similar

Reply:
👍